azurelinuxagent/ga/exthandlers.py

```diff
@@ -510,27 +522,20 @@ def handle_ext_handlers(self, goal_state_id):
         depends_on_err_msg = None
         extensions_enabled = conf.get_extensions_enabled()
 
-        # Instantiate policy engine, and use same engine to handle all extension handlers. If an error is thrown during
-        # policy engine initialization, we block all extensions and report the error via handler status for each extension.
-        # Save policy to history folder.
+        # Read policy file and update policy in engine. If an error is thrown during policy update, we block all extensions
+        # and report the error via handler status for each extension.
         policy_error = None
         try:
-            gs_history = self.protocol.get_goal_state().history
-            policy_engine = ExtensionPolicyEngine()
-            if policy_engine is not None and policy_engine.policy_file_contents is not None and gs_history is not None:
-                gs_history.save(policy_engine.policy_file_contents, "waagent_policy.json")
-
+            self._policy_engine.update_policy(self.protocol.get_goal_state().history)
```

Comment:
this should be done at the very top of ExtHandlersHandler.run, along with the intialization of  self.ext_handlers